src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,50 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] stack;
+    private int size;
 
+    public MateStack() {
+        stack = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public void push(T value) {
+        stack[size++] = value;
+        resize();
     }
 
     public T peek() {
-        return null;
+        return getValue();
     }
 
     public T pop() {
-        return null;
+        T oldValue = getValue();
+
+        T[] oldStack = stack;
+        stack = (T[]) new Object[size];
+        System.arraycopy(oldStack, 0, stack, 0, --size);
```

Comment:
If there is no need to trim the stack this part is redundant, unless you want to keep it